scripts/cluster_network.py

```diff
@@ -420,19 +420,22 @@ def fix_country_assignment_for_hac(n):
                     component == component_sizes.index[-1]
                 ].index[0]
 
-                neighbor_bus = n.lines.query(
```

Comment:
Nice check!
If the problem is the allocation of the country feature, can't there be a better solution? for example storing it into the n.buses before?
In the case of CD I fear that this trick would allocate the bus to the wrong country if there is a close bus on the other border.

Isolated buses are likely at remote areas, thus the allocation may be improper

Reply:
I also feel that it might be worth to re-implement the county-assignment part keeping in mind difference in the processed data

Reply:
Thanks for the discussion today.
We revised the PR and I think the latest fix with the revise of the lines here referred to is appropriate.
It may be of interest also for pypsa-eur (@fneum)
I think this review is almost ready; latest checks before merging